lib/ain-grpc/src/rpc/eth.rs

```diff
@@ -452,11 +469,7 @@ impl MetachainRPCServer for MetachainRPCModule {
         block_number: BlockNumber,
         full_transactions: Option<bool>,
     ) -> RpcResult<Option<RpcBlock>> {
-        let block_number = self.block_number_to_u256(Some(block_number))?;
-        debug!(target:"rpc", "Getting block by number : {}", block_number);
-        self.handler
-            .storage
-            .get_block_by_number(&block_number)
+        self.get_block(Some(block_number))
```

Comment:
Use fallible try_get_block method instead here?

Reply:
Correction - get_block is fallible, we should do a proper method try_get_block_from_number, and implementers should propagate the block error accordingly if the block query fails from the db